src/SessionManager/ProgramSettings.cc

```diff
@@ -76,7 +76,8 @@ ProgramSettings::ProgramSettings(int argc, char** argv) {
  3   warning
  4   info
  5   debug
- 6   trace)");
+ 6   performance
+ 7   protocol)");
```

Comment:
Swap 6 and 7, and I suggest changing "protocol" to something like "+ protocol messages", otherwise users will expect a level called protocol in the logs, which we don't have.

Reply:
I think the only way to make this consistent is to move it to a separate input argument `debug_protocol_messages` or something like that, and have that displayed using `spdlog::debug` with an additional prefix `[protocol]`

Reply:
That would leave us with something like 
`[2021-02-25 11:04:26.647] [debug] [protocol] <== REGISTER_VIEWER`

Reply:
I think that we should use a separate flag for protocol messages *and* performance messages, and log performance messages to `info`, instead of using the `trace` level for them.

So

* `log_performance` -> performance logged to `info` with added `[performance]` prefix
* `log_protocol_messages` -> ICD messages logged to `debug` with added `[protocol]` prefix

Reply:
I'm happy with that suggestion. However, I do think elegantly adding a `[performance]` prefix might be a little tricky, since we need to handle variadic arguments. 

@markccchiang please give this a try